tools/define-compose-url.sh

```diff
@@ -1,7 +1,10 @@
 #!/bin/bash
 set -euo pipefail
 
-curl -L http://download.devel.redhat.com/rhel-8/nightly/RHEL-8/latest-RHEL-8.4/COMPOSE_ID > COMPOSE_ID
+# Small upgrade, this will get the latest nightly COMPOSE_ID with the specified VERSION
+curl -L "https://cts.engineering.redhat.com/api/1/composes/?release_version=${VERSION}&tag=nightly" | jq -r .items[0].compose_info.payload.compose.id > COMPOSE_ID
```

Comment:
Where does the `VERSION` variable come from? Seems like it is undefined to me?

Reply:
FTR this will return `null` once the nightly is removed (e.g. old enough). Maybe we should error out explicitly if such condition is met.   Which isn't much different than relying on the symlink URL in the old version and handling the errors gracefully. 

What's the purpose of this change? How does that help us solve our problems with missing symlink or missing nightly trees  ?

Reply:
The `VERSION` variable comes from the scheduled pipeline settings, you can set additional env vars just for that run. The idea is to add stages that would only be ran as a scheduled nightly test. So when the nightlies will stop existing, we can just bump the variable version in gitlab.

This is still a WIP, I don't have everything figured out yet. I'm mostly trying if this approach will work for us and trying out some things along the way. This should hopefully point to the latest working nightly, but when the time comes where they're going to be old enough to not exist anymore and they also stop building them I don't yet know how to automatically figure that out...

Reply:
> What's the purpose of this change? How does that help us solve our problems with missing symlink or missing nightly trees ?

I guess that when the time when nightly builds are no longer available we'll have to manually bump the `VERSION` and probably also change the aws runners. So what this could do for us is this:

1. We'll always use latest finished and accepted nightly compose
2. Pointing to a newer minor build is as easy as changing the `VERSION` in gitlab pipeline schedules

Currently I can't think of a better way. In case this stops working because of missing nightly trees/symlink it means it's time to drop the current nightly testing and bump the version. Also it won't return `null` but the latest tagged nightly `COMPOSE_ID`. Of course that means that if the build is too old, the `COMPOSE_URL` won't exist anymore...

If you think that this won't help us I'll try to come up with something different..

Reply:
Few comments FTR:

> The VERSION variable comes from the scheduled pipeline settings, you can set additional env vars just for that run. > The idea is to add stages that would only be ran as a scheduled nightly test. 

That kind of defeats the idea of configuration as code!

> So when the nightlies will stop existing, we can just bump the variable version in gitlab.

We can do the same in git and have traceability of the changes so it doesn't really solve the problem just shifts the configuration from one place to another.

> I guess that when the time when nightly builds are no longer available we'll have to manually bump the VERSION and probably also change the aws runners. So what this could do for us is this:
>
>    We'll always use latest finished and accepted nightly compose
>    Pointing to a newer minor build is as easy as changing the VERSION in gitlab pipeline schedules
>
> Currently I can't think of a better way. In case this stops working because of missing nightly trees/symlink it means it's time to drop the current nightly testing and bump the version. Also it won't return null but the latest tagged nightly COMPOSE_ID. Of course that means that if the build is too old, the COMPOSE_URL won't exist anymore...

I don't think this is any different from what we have ATM. It just shifts the definitions/variables into a different place. 

With this PR let's focus on making the nightly pipeline run more or less in the same way it does now and solve the remaining problems separately.

Reply:
Yeah, you're right, I see it now.
I've put in some default values but I'd like to keep the option of specifying the `VERSION` and `TAG` from outside, I believe it can be useful if we want to run the tests on some other compose. 

Also one thing that's an improvement over using the latest symlink is that we're really using the latest compose that's finished and passed acceptance testing. I could see a newer compose in the 8.5 latest folder but that was actually unusable. I'm going to open a ticket for that so it won't happen again but using this internal CTS service sounds safer to me.

I'm going to try running the whole test suite using RHEL 8.4 runner, if that passes I think we should be in the same state as with Jenkins. Scheduled pipeline will be our cron trigger and the `prepare-rhel-internal` part will only run during the scheduled pipeline run. 

Also I think we should get 8.5 runner in aws, I'm going to work that out with Ondrej.

Reply:
> I've put in some default values but I'd like to keep the option of specifying the VERSION and TAG from outside, I believe it can be useful if we want to run the tests on some other compose.

Again, how is `VERSION` & `TAG` any different from specifying the `COMPOSE_URL` directly?  It does the same thing but adds a few more steps in the process. If you want to test something different then you can specify the URL to SUT. 

IMO none of these changes are needed or beneficial at the moment.

Reply:
Right, I've removed both `VERSION` and `TAG` it's simpler to just provide the whole `COMPOSE_URL`.